### DISCUSSION TITLE
403 error when Dependabot tries to create a pull request

### DISCUSSION THREAD

**Discussion by semkeijsper** [mburumaxwell/paklo#1245] [Q&A] [answered]
**Describe the bug**
Dependabot fails when trying to create a pull request

**To Reproduce**
1. Run the pipeline with the Dependabot task
2. The plugin access the code and checks all the packages
3. It fails when trying to create a pull request

**Expected behavior**
It should create a pull request

**Screenshots**
![image](https://github.com/tinglesoftware/dependabot-azure-devops/assets/78076164/0c3fc37a-e167-4b7f-9ec0-0da70f8db026)

**Extension (please complete the following information):**
 - Host: Azure DevOps
 - Version: 1.28.0.708 (Latest)

**Server (please complete the following information):**
 - Region: westeurope
 - Version: ?

**Additional context**
The plugin can access the code in the repository, so the API connection works. It can also do GET requests for fetching the existing pull requests. So the basic permissions work, but as soon as it tries a POST request, it fails.
We have tried all different options but cannot find where we can configure the permissions for this plugin, or for the 'user' that runs this plugin. The documentation of this plugin is unclear about where we can configure the required permissions.
We use minimal configs for dependabot and we use the standard Azure Pipelines agent pool. Please advice on where this goes wrong.

**Comment by ahmed-ben10** [comment]
I encountered a similar issue just a few days ago. To resolve it, ensure that your build service user has been granted the following permissions:

![image](https://github.com/tinglesoftware/dependabot-azure-devops/assets/44867080/9f5b7430-3367-42fc-a8b7-bd17aad3db2a)

**Comment by semkeijsper** [comment]
@ahmed-ben10 Thanks for your reply. How did you discover/find who your build service user is? And in what menu was that screenshot taken, Project settings, Organization settings, and which submenu? Thanks.

**Comment by sambouchard-newforma** [comment]
> @ahmed-ben10 Thanks for your reply. How did you discover/find who your build service user is? And in what menu was that screenshot taken, Project settings, Organization settings, and which submenu? Thanks.

To find who your Build service user is, go to the dropdown menu on the right in the pipeline page and click on ''Manage security'' : 
![image](https://github.com/tinglesoftware/dependabot-azure-devops/assets/149517997/a3232e8d-2565-4b4e-bf1a-52c4f358712b)
Here at the bottom of the list you will see under users who your build service user(s) is/are :

Then go to the permissions tab of the Azure DevOps project settings in the users section you will most likely find the user that you found previously. From here on, you can either modify its permission or add the user to be the member of a group that has the correct permissions!

**Comment by semkeijsper** [comment]
@sambouchard-newforma Unfortunately none of the users listed in the Pipeline dropdown can be found in the Permissions > Users tab... thanks for the reply though :)

**Comment by mburumaxwell** [comment]
Check https://github.com/tinglesoftware/dependabot-azure-devops/issues/1095#issuecomment-2141089531

**Comment by rdehouss** [comment]
Hi everyone,
I'm having exactly the same issue (403 on pushes api) using dependabot extension 1.29.769

I tried to give my "Build service" user and Project Collection Build Service Accounts the mentioned permission without success :
![image](https://github.com/user-attachments/assets/864c085a-d186-4bac-ae71-f8e927eb1654)
![image](https://github.com/user-attachments/assets/fa3e4543-c8d1-4fa8-b555-99b4da158134)
https://github.com/dependabot/dependabot-core/blob/main/common/lib/dependabot/clients/azure.rb#L183C11-L183C24

With my user PAT in inputs: azureDevOpsAccessToken, it works fine.

Any other idea?

Thanks in advance!

Cheers,
Raphaël

**Comment by rdehouss** [comment, marked as answer]
Finally, I could make it work by following this article : https://medium.com/@sugam.arora23/automated-dependency-management-harnessing-dependabot-for-seamless-updates-on-azure-devops-ea5d1a7aa82c
Basically, I gave the permissions to the user (search for it) "Project Collection Build Service (<your org>)" on the security tab on All repositories
![image](https://github.com/user-attachments/assets/4c528bd1-70e1-4a98-b5d2-39585acbb4ee)

**Reply by semkeijsper** [reply]
This is what ended up working for me, thanks!

**Comment by thepaulmacca** [comment]
I'm seeing a completely new set of permissions here now

![image](https://github.com/user-attachments/assets/73603969-4fe5-42e0-a1ed-bd95dc72684b)
